docs/ref/templates/language.txt

```diff
@@ -431,6 +431,104 @@ it also defines the content that fills the hole in the *parent*. If there were
 two similarly-named :ttag:`block` tags in a template, that template's parent
 wouldn't know which one of the blocks' content to use.
 
+.. _recursive-inheritance:
+
+Recursive inheritance
+---------------------
+
+While not clear from the above examples, :ttag:`{% extends %}<extends>` can be
+used to inherit and extend from a template of the same name. That template in
+turn can extend another template of the same name. This semi-recursive pattern
+continues until reaching a template that doesn't extend just like normal
+inheritance. This isn't true recursion, as the same template file isn't used
+more than once, instead this is just another way of using inheritance to build
+layers into a single template that can be reused by multiple other templates.
+
+The best way to understand this use of inheritance is with a simple example:
+
+.. snippet::
+    :filename: reusable_app/templates/reuseable_app_base.html
+
+    {% block header %}
+    <h1>Title from the reusable app</h1>
+    {% endblock %}
+
+    {% block main %}
+    <p>A paragraph in the reusable app.</p>
+    {% endblock %}
+
+    {% block footer %}
+    <footer>Footer text from the reusable app</footer>
+    {% endblock %}
+
+Here we have a reusable app that gives us a ``reuseable_app_base.html``
+template. Let's imagine for this example that this is a useful third-part app,
```

Comment:
`third-party` instead of `third-part`, I think?

Reply:
Thanks! Fixed.